### PR TITLE
slitspatnum bug

### PR DESCRIPTION
**Pull request by jhennawi**
It looks like the functionality to reduce specific slits was never properly implemented. It apparently will fail on any instrument that does not have the spectrograph.list_detectors() method overloaded which is all instruments except gemini_gmos, keck_deimos, and keck_mosfire.

[rdx]
  slitspatnum = DET01:1557

The current PR attempts to fix this using the specific case of LBT mods where the failure was identified.  I need help with Keck hires, since the mosaic hooks are not there and I'm confused with how this function is supposed to behave since the list_detectors function is different for the no mosaic case for deimos and mosfire (I'm not sure why). 

See the dev-suite PR of the same name for a test example.  I'm handing off this PR to @kbwestfall and @debora-pe who I think are in a better position to understand whether my changes are correct. Also, please note that I think that list_detectors should maybe be overloaded for keck_hires which also has mosaics, but I'm a bit unclear on that.

It would also be a good idea to add tests.

### CODE REVIEW

**Comment by kbwestfall** [comment]
Hi @jhennawi .  I made some changes to the branch and now 
```
run_pypeit lbt_mods_mods1r_longslit.pypeit
```
finishes successfully.  I get a `spec1d_mods1r.20180112.0051-QSO_J1104+2134_MODS1R_20180112T113720.755.txt` file that looks like this:

```
| slit |                    name | spat_pixpos | spat_fracpos | box_width | opt_fwhm |  s2n | wv_rms |
| 1557 | SPAT1643-SLIT1557-DET01 |      1643.5 |        0.677 |      3.00 |    1.224 | 0.31 |  0.170 |
```

Is that what you're expecting?  If so, I can run the `mods_bug` dev-suite branch on this pypeit branch.

File: pypeit/tests/test_spectrographs.py [hunk 1]
```diff
@@ -62,6 +62,44 @@ def test_select_detectors_mosaic():
...
+    assert dets.size == 1, 'MODS1R has 1 detector'
+    with pytest.raises(PypeItError):
+        mosaics = mods.list_detectors(mosaic=True)
+
+
+def test_list_detectors_hires():
+    hires = load_spectrograph('keck_hires')
+    dets = hires.list_detectors()
+    assert dets.ndim == 1, 'HIRES has a 1D array of detectors'
+    assert dets.size == 3, 'HIRES has 1 detector'
```
Comment (debora-pe) [inline pypeit/tests/test_spectrographs.py]: Did you want to say "HIRES has 3 detectors"?

**Review by debora-pe — APPROVED** [review]
Thank you @kbwestfall for taking care of this quickly and @jhennawi for starting this PR.
Just for the records,  `list_detectors` was first introduced to be used for slitmask design purposes (as the docstring says). This is why it is different for DEIMOS (we wanted to list red and blue detectors separately). I'm glad it can now be used for a broader purpose.

File: pypeit/tests/test_spectrographs.py (hunk 1 above)
Comment (kbwestfall) [reply pypeit/tests/test_spectrographs.py]: Yes!  Good catch.  Fixed.

**Review by kbwestfall — APPROVED** [review]

**Pull request merged**